apps/total-typescript/src/pages/api/sanity/transcript/update.ts

```diff
@@ -0,0 +1,117 @@
+  req: NextApiRequest,
+  res: NextApiResponse,
+) => {
+  if (req.method === 'POST') {
+    const client = sanityClient({
+      projectId: 'z9io1e0u',
+      dataset: 'production',
+      useCdn: process.env.NODE_ENV !== 'development', // `false` if you want to ensure fresh data
+      apiVersion: '2021-10-19',
+      token: process.env.SANITY_API_TOKEN,
+    })
+
+    if (req.body.event === 'TRANSCRIPT_COMPLETE') {
+      const {audiofile} = req.body
+
+      const transcript = await fetch(
+        `https://castingwords.com/store/API4/audiofile/${audiofile}/transcript.txt?api_key=${process.env.CASTINGWORDS_API_TOKEN}`,
+      )
+        .then((transcript) => {
+          return transcript.text()
+        })
+        .catch((err) => {
+          console.log(err)
+          res.status(500).json({
+            message: 'Failed to fetch transcript .txt file from Castingwords',
+          })
+          return
+        })
+
+      const srt = await fetch(
+        `https://castingwords.com/store/API4/audiofile/${audiofile}/transcript.srt?api_key=${process.env.CASTINGWORDS_API_TOKEN}`,
+      )
+        .then((srt) => {
+          return srt.text()
+        })
+        .catch((err) => {
+          console.log(err)
+          res
+            .status(500)
+            .json({message: 'Failed to fetch .srt file from Castingwords'})
+          return
+        })
+
+      const getDocumentQuery = `
+        *[
+          _type == "videoResource" &&
+          castingwords.audioFileId == "${audiofile}"
+        ] {_id, castingwords}
+        `
+
+      client
+        .fetch(getDocumentQuery)
+        .then((docs: any) => {
+          const {_id, castingwords} = docs[0]
+          client
+            .patch(_id)
+            .set({
+              castingwords: {
+                ...castingwords,
+                transcript: [
+                  {
+                    style: 'normal',
+                    _type: 'block',
+                    children: [
+                      {
+                        _type: 'span',
+                        marks: [],
+                        _key: uniqueId('body-key-'),
+                        text: transcript,
+                      },
+                    ],
+                    markDefs: [],
+                    _key: uniqueId('block-key-'),
+                  },
+                ],
+                srt: srt,
+              },
+            })
+            .commit()
+            .then(() => {
+              res.status(200).json({message: 'Video resource updated'})
+              return
+            })
+            .catch((err) => {
+              console.log(err)
+              res.status(500).json({message: 'Failed to update video resource'})
+              return
+            })
+        })
+        .catch((err) => {
+          console.log(err)
+          res.status(500).json({
+            message: `Invalid query for Sanity document`,
+          })
+        })
+    } else {
```

Comment:
is this invalid or just something we don't handle? we should probably handle it accordingly if it's just a status not being handled.

Reply:
This error is thrown when it can't find a document with the audiofile id. I can update the message to be more specific to that